src/Aspire.ProjectTemplates/templates/aspire-starter/AspireStarterApplication.1.Tests/AspireStarterApplication.1.Tests.csproj

```diff
@@ -1,3 +1,28 @@
+<!--#if (TestProjectFrameworkKind == "MSTest") -->
+<Project Sdk="MSTest.Sdk/3.3.1">
```

Comment:
This is very unusual to see for me. I'm not aware of any templates we ship today that use a versioned SDK like this. Are we fully aware of the user experience implications of this?
/Cc @mhutch @dsplaisted

Reply:
That's fresh off the press https://devblogs.microsoft.com/dotnet/introducing-mstest-sdk/

Reply:
Sure but I'm more speaking to the implications of having a version in the Sdk identifier and relying on the NuGet SDK resolver. Those have existed for years now but not in wide use as we don't ship templates using them. Are there any interactions that change the user experience because of that change, e.g. what happens if I'm using a custom NuGet config that doesn't include nuget.org (as per our secure supply chain guidelines)? Does project load fail in a nice way in Visual Studio if the SDK NuGet package cannot be restored? Putting this in the Aspire workload means that many more users will be exposed to these behaviors so I want to be very sure we know what the experience is and are happy with it.

Reply:
@baronfel What's your view about risks?

@DamianEdwards We have an option without the SDK but if we can clear out any risk, we would prefer to push the SDK forward as it simplifies many things. For example, for v3.4, we can introduce `EnableAspire` property that would add the using of `Aspire.Hosting.Testing` and the package reference `Aspire.Hosting.Testing`.

Reply:
> Are there any interactions that change the user experience because of that change, e.g. what happens if I'm using a custom NuGet config that doesn't include nuget.org (as per our secure supply chain guidelines)?

What you mean? The test framework packages and the test sdk comes from nuget, the same issue should happen with the normal configuration. The SDK is only a way to simplify users setup.

Reply:
The problem with resolving MSBuild SDKs from NuGet is that they cause project evaluation to block on resolving the package from NuGet, and AFAIK Visual Studio does not handle this gracefully, so it can cause the UI to hang.

Reply:
I had an offline sync with @rainersigwald and VS at the moment does have some suboptimal usage of the MSBuild sdk in some scenario, if we will have users complain about that we will work together with the MSBuild/VS team for improve the situation.

Reply:
> if we will have users complain about that we will work together with the MSBuild/VS team to improve the situation.

While I think this is a fine stance for the MSTest team to take, I'm not comfortable right now having the Aspire product be the "guinea pig" to flesh out the experience issues so I'd prefer we stick to the standard Test SDK for these templates. We can revisit this for the next major version of Aspire (releasing with .NET 9).

Reply:
@Evangelink 
> For example, for v3.4, we can introduce EnableAspire property that would add the using of Aspire.Hosting.Testing and the package reference Aspire.Hosting.Testing.

Auto-adding package references from SDKs is not something we typically like doing, as it hides the version from customers and in this case would mean your SDK has to keep track of Aspire package versions which seems like a bad dependency to create. This is why the Aspire workload SDK doesn't auto-add the `Aspire.Hosting.AppHost` package to AppHost projects (those with `<IsAspireHost>true</IsAspireHost>` set in the csproj).

Reply:
> While I think this is a fine stance for the MSTest team to take, I'm not comfortable right now having the Aspire product be the "guinea pig" to flesh out the experience issues so I'd prefer we stick to the standard Test SDK for these templates. We can revisit this for the next major version of Aspire (releasing with .NET 9).

Fine by me, I'll update the PR.

Reply:
Resolving this thread, I am now using MSTest metapackage and no longer the SDK package.